src/menu.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use glib_ffi;
+use glib::object::Downcast;
+use glib::translate::*;
+use Menu;
+use IsA;
+use Widget;
+use std::boxed::Box as Box_;
+use std::mem::transmute;
+use libc::c_int;
+
+impl Menu {
+    pub fn popup<T: IsA<Widget>, F: Fn(&Self, &mut i32, &mut i32, &mut bool) + 'static>(
+        &self, parent_menu_shell: &T, parent_menu_item: &T, f: F, button: u32, activate_time: u32) {
```

Comment:
About the button argument: I think it's supposed to be an enum (an event one as it seems) but cannot find it. One of its member is supposed to be RIGHT_CLICK but still not able to find it. Any idea?